x/mailbox/keeper/keeper.go

```diff
@@ -45,20 +48,33 @@ func NewKeeper(cdc codec.BinaryCodec, key storetypes.StoreKey, cwKeeper *cosmwas
 		authority:   authority.String(),
 		mailboxAddr: authtypes.NewModuleAddress(types.ModuleName),
 		version:     0,
-		domain:      domain,
 		pcwKeeper:   cosmwasm.NewDefaultPermissionKeeper(cwKeeper),
 		Tree:        &imt.Tree{},
 		Delivered:   map[string]bool{},
 	}
 }
 
-func (k Keeper) VerifyMessage(messageBytes []byte) (string, error) {
+func (k *Keeper) SetDomain(c context.Context, domain uint32) {
```

Comment:
It is still unclear to me why the domain needs to be in the KVStore. It adds r/w operations.

Reply:
Without it in the genesis file, why doesn't it persist in memory when other items like the authority do?